### PR TITLE
Add filtering by health and state in list rules

### PR DESCRIPTION
**Pull request by euniceek**
**What this PR does**:

This PR adds two additional filtering fields to the list rules API. If users want to list rules that are in firing state or healthy, they will be able to do it using the following parameters.

- `api/v1/rules?state=firing`
- `api/v1/rules?health=ok`

**Checklist**
- [X] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ruler/api.go
Comment (yeya24) [inline pkg/ruler/api.go:138]: Do we need `strings.TrimSpace`? We don't do it for other parameters.

Comment (euniceek) [reply pkg/ruler/api.go:138]: Yep, removing since other params doesn't do it.

**Review by yeya24 — APPROVED** [review]
It looks good to me. Thanks!

**Comment by yeya24** [comment]
@euniceek Can you please update the PR description? I don't think Cortex supports ruleinfo API

**Comment by euniceek** [comment]
> @euniceek Can you please update the PR description? I don't think Cortex supports ruleinfo API

Thanks for the review! Updated!

**Comment by yeya24** [comment]
Let's wait for another approval from a maintainer before merging this.

**Comment by yeya24** [comment]
@euniceek Can you also update the changelog? We can call it an `enhancement`

**Comment by alanprot** [comment]
ok.. LGTM!

**Review by alanprot — APPROVED** [review]

**Pull request merged**
